src/main/java/seedu/address/commons/core/index/UniqueIdGenerator.java

```diff
@@ -32,6 +36,35 @@ public UniqueId next() {
         return new UniqueId(id);
     }
 
+    /**
+     * Adds a unique id to the set of storedIdPet
+     */
+    public static boolean addToStoredIdPet(UniqueId id) {
+        return storedIdPet.add(id);
+    }
+
+    /**
+     * Checks if the storedIdPet set contains a given id.
+     */
+    public static boolean storedIdPetContains(UniqueId id) {
+        return storedIdPet.contains(id);
+    }
```

Comment:
Is it better if this method is private and only called by next()? I mean each time next() is called, next() will check if the next unique id is in the set.

Reply:
Yea I think that would be better. I will consider to change that after fixing other bugs in the issue tracker